### ISSUE TITLE
Simplify network messages for requests and events

### ISSUE THREAD

**Issue by lyonsil** [paranext/paranext-core#145]
Discussed in https://docs.google.com/document/d/1caYfnVs15lP9bpWLZG6e3nYgwAF5zZWUAv6o2TuoSY0/edit#heading=h.s7q79wllkfhc

1) Scrap the "object", "server", and "command" scope qualifiers for request types and add a new, optional JSON property called "networkObject".  The purpose of "networkObject" is to define who handles the request.  The value of "networkObject" will be the name of a network object containing the function specified in "requestType".  If the "requestType" is not intended to route to a network object but instead be handled by the server or a registered command, then "networkObject" should not be supplied.

Example requests in the new layout might be:
```
{"type":"request","requestType":"registerNetworkObject","requestId":5,"contents":["calculator"],"senderId":2}
{"type":"request","requestType":"add","networkObject":"calculator","requestId":123,"contents":[1, 2],"senderId":6}
```

2) Scrap the "network" and "object" scope qualifiers for events and create "eventSource".  Then people can subscribe to events based on the source and type of events.  The payload type depends on the source and type.  The source will either be "server" or a network object name.

Example events in the new layout might be:
```
{"type":"event","eventType":"onDidClientConnect","eventSource":"server","senderId":0,"event":{"clientId":2,"didReconnect":false}}
{"type":"event","eventType":"dataUpdated","eventSource":"someNetworkObject","senderId":4,"event":""}
```

**Comment by lyonsil** [comment]
If/when we do this, let's consider adding serialization checks whenever we're given an object that we have to `JSON.stringify`.  We don't want to end up serializing something that isn't going to deserialize properly or not round-trip properly.

**Comment by tjcouch-sil** [comment]
Doing some discussion about this on https://docs.google.com/document/d/1sfAFcPGDUJA9YaAkPeC6A0q_jASfyNLIkHs9CDIF4mQ/edit

**Comment by lyonsil** [comment]
Resolved by #1200
